units.md

```diff
@@ -0,0 +1,86 @@
+# Massa Units
+
+**Authors:**
+
+**Status:** Draft
+
+**Version:** 0.1
+
+## Abstract
+
+This specification defines the units used to measure the different objects of the Massa blockchain. It aims to provide a standardized format for the different units to offer the simplest and most readable user experience possible while being consistent from one application to another within our ecosystem.
+
+> _NOTE:_ The manipulation of very large or very small numbers will be simplified by the use of [metric prefixes](https://en.wikipedia.org/wiki/Metric_prefix) of the [SI](https://en.wikipedia.org/wiki/International_System_of_Units).
+
+## Targeted Audience
+
+This specification is intended for developers who wish to interact with the Massa blockchain network using various tools and applications such as node API, smart contract tooling, dApps tooling, Thyra, and Thyra plugins.
+
+## Units
+
+### Massa coin
+
+Massa coin is the name of the native coin on the Massa blockchain.
+Massa coin is expressed as an unsigned 64-bit integer.
+
+1 Massa is equivalent to 1,000,000,000 nanoMassa, the smallest unit of Massa coin.
+
+To simplify usage, the following standard units implemenation are recommended:
+
+- milliMassa (mMASSA): 1 milliMassa represents 1,000,000 nanoMassa and also 1/1000th of a Massa.
+- microMassa (µMASSA): 1 microMassa represents 1,000 nanoMassa and also 1/1,000,000th of a Massa.
+- nanoMassa (nMASSA): 1 nanoMassa represents the smallest unit of Massa coin and also 1/1,000,000,000th of a Massa.
+
+### Gas
+
+Gas is a virtual unit used to measure the computational complexity of executing transactions on the Massa blockchain. Each operation requires a certain amount of Gaz, and the amount of Gaz consumed is proportional to the computational complexity of the operation. The unit of gas is calibrated to be equivalent to 300 milliseconds of execution time on the target hardware of the sum of MAX_ASYNC_GAS and MAX_BLOCK_GAS.
+Gaz is expressed as an unsigned 64-bit integer.
+
+1 gas unit (GAS) is equivalent to 1,000,000,000,000 picoGas, the smallest unit of gaz.
+
+To simplify usage, the following standard units implemenation are recommended:
+
+- milliGas (mGAS): 1 milliGas represents 1,000,000,000 picoGas or 1/1000th of a gas unit.
+- microGas (µGAS): 1 microGas represents 1,000,000 picoGas or 1/1,000,000th of a gas unit.
+- nanoGas (nGAS): 1 nanoGas represents 1,000 picoGas or 1/1,000,000,000th of a gas unit.
+- picoGas (pGAS): 1 picoGas represents the smallest unit of gaz or 1/1,000,000,000,000th of a gas unit.
```

Comment:
As @sebastien-forestier pointed out we have blocks of 1,000,000,000 units of gas so if I understand correctly GAS should be the smallest unit and it would be more useful to have kilo, mega ...

Reply:
@Ben-Rey let's Seb come back to us on this. I want to finish this PR ASAP: we are loosing time on secondary subject where we should focus on priority.

Reply:
Yes gas is currently the smallest unit, which can't be fractioned.
Personally I don't think that it is a good idea to recommend using pico/nano/micro or kilo/mega/giga etc, why not just use the basic unit with no conversion ? Now it's true we can propose them and builders decide if they use them, but I don't think it simplifies, it adds conversions and potential confusion.
I would use thousands, millions and billions to talk about big gas quantities

Reply:
Because it's simply unreadable for the Dev, to much 0.